File: Data Structures & Algorithms/two-integer-sum/submission-1.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let map = nums
            .iter()
            .enumerate()
            .map(|(idx, num)| (*num, idx as i32))
            .collect::<HashMap<i32, i32>>();
        for (idx, num) in nums.into_iter().enumerate() {
            let c = target - num;
            if let Some(idxx) = map.get(&c) {
                if *idxx != idx as i32 {
                    return vec![
                        std::cmp::min(*idxx, idx as i32),
                        std::cmp::max(*idxx, idx as i32),
                    ];
                }
            }
        }
        todo!()
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles) + 1

        ans = l + r
        while l < r:
            m = (l + r) // 2
            sm = 0
            for pile in piles:
                if pile % m == 0:
                    sm += pile // m
                else:
                    sm += int(pile / m) + 1
                
            if sm <= h:
                ans = min(ans, m)
                r = m
            else:
                l = m + 1
        return ans
